### DISCUSSION TITLE
New foreign and id methods for migrations

### DISCUSSION THREAD

**Discussion by bakle** [laravel/framework#47336] [Ideas]
This proposal is to follow the same approach of `$table->id()` and `$table->foreignId()` by adding new syntactic sugar methods for:
* tinyIncrements
* smallIncrements
* mediumIncrements

And include their respective foreign methods for each one.

## Proposal
Let's say we have three tables, `states`, `addresses` and  `warehouses`, where `addresses` table has a foreign key of the `states` table, and `warehouses` has a foreign key of the `addresses` table.

```php
Schema::create('states', function (Blueprint $table) {
    $table->tinyId();
   ...
});
```

```php
Schema::create('addresses', function (Blueprint $table) {
    $table->smallId();
    $table->tinyForeignId('state_id')->constrained('states');
    ...
});
```

```php
Schema::create('warehouses', function (Blueprint $table) {
    $table->mediumId();
    $table->smallForeignId('address_id')->constrained('addresses');
    ...
});
```

**Comment by cosmastech** [comment]
I had a similar idea for this and just put up a PR https://github.com/laravel/framework/pull/47346
